pgmpy/inference/mplp.py

```diff
@@ -0,0 +1,752 @@
+import itertools
+import numpy as np
+from operator import sub
+from operator import add
+import queue
+import math
+from collections import namedtuple
+from pgmpy.models import MarkovModel
+from pgmpy.inference import ApproxInference
+
+
+class MPLP(ApproxInference):
+    """
+    Class for performing inference using Max-Product Linear Programming Relaxations.
+    We use the Max-Product Linear Programming (MPLP) algorithm to minimize the dual .
+    """
+
+    def __init__(self, model):
+        super().__init__(model)
+
+        if not isinstance(model, MarkovModel):
+            raise ValueError('The model given is not MarkovModel.')
+
+        # For containing the cardinalities of all the variables present in the model
+        self.m_var_sizes = self.cardinality
```

Comment:
Don't use ```m_var_name``` in ```Python```. It is a C++ style used to setting private variables; we don't require over here. So please remove ```m_``` from the start of all the variables.